src/server/game/Spells/Spell.cpp

```diff
@@ -5498,7 +5498,7 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
             case SPELL_AURA_MOUNTED:
             {
-                if (m_caster->IsInWater())
+                if (m_caster->IsInWater() && m_spellInfo->HasAura(SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED))
```

Comment:
I would agree with @Takenbacon. I think that this patch is only valid for 6.x.x branch and not for 3.3.5 patch but I may be wrong again.